src/libraries/System.Private.Xml/src/System/Xml/XmlConvert.cs

```diff
@@ -693,6 +693,18 @@ public static string ToString(int value)
             return value.ToString(null, s_invariantCultureInfo);
         }
 
+        internal static void AppendInt(ref ValueStringBuilder vsb, int value)
+        {
+            Span<char> tmpSpanDestination = stackalloc char[11];
+            if (value.TryFormat(tmpSpanDestination, out int charsWritten, provider: s_invariantCultureInfo))
+            {
+                var vsbSpan = vsb.AppendSpan(charsWritten);
+                tmpSpanDestination.Slice(0, charsWritten).CopyTo(vsbSpan);
+            }
+            else
+                vsb.Append(value.ToString(null, s_invariantCultureInfo));
+        }
```

Comment:
You can move:
https://github.com/dotnet/runtime/blob/d936a661c56f742b1660279aacb9a47d4ba1bd30/src/libraries/System.Private.CoreLib/src/System/Text/ValueStringBuilder.AppendFormat.cs#L46-L56
into:
https://github.com/dotnet/runtime/blob/d936a661c56f742b1660279aacb9a47d4ba1bd30/src/libraries/Common/src/System/Text/ValueStringBuilder.cs
and then just use that.

Reply:
@stephentoub This look like a pretty big/general change and I do not prefer to do it here. Do you think if is is a problem to add partial class only in this project with the same name of the method, so if somebody is brave enough and do the proposed change - will delete my class as it will fail in compile time ?